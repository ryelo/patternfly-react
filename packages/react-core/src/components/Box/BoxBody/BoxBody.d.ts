import { HTMLProps } from 'react';

export interface BoxProps extends HTMLProps<HTMLDivElement> {
  children?: React.ReactNode;
}

declare const BoxBody: React.SFC<BoxProps>;

export default BoxBody;
