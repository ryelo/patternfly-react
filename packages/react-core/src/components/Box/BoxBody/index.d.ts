export {
  default as BoxBody,
  BoxBodyProps
} from './BoxBody';
