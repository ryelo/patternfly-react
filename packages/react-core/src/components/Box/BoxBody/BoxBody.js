import React from 'react';
import { css } from '@patternfly/react-styles';
import PropTypes from 'prop-types';
import styles from '@patternfly/patternfly-next/components/Box/styles.css';

const propTypes = {
  /** content rendered inside the boxBody */
  children: PropTypes.node,
  /** additional classes added to the boxBody */
  className: PropTypes.string
};

const defaultProps = {
  children: '',
  className: ''
};

const BoxBody = ({ children, className, ...props }) => (
  <div {...props} className={css(styles.boxBody, className)}>
    {children}
  </div>
);

BoxBody.propTypes = propTypes;
BoxBody.defaultProps = defaultProps;

export default BoxBody;
