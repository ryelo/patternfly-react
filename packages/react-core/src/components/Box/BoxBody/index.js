export { default as BoxBody } from './BoxBody';
