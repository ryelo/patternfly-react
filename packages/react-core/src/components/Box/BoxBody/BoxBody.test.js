import BoxBody from './BoxBody';
import React from 'react';
import { shallow } from 'enzyme';

test('Box has body', () => {
  const view = shallow(<BoxBody>Box Body</BoxBody>);
  expect(view).toMatchSnapshot();
});
