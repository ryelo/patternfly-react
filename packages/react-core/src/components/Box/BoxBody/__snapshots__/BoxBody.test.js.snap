// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box has body 1`] = `
.pf-c-box__body {
  display: block;
  padding: 2rem 2rem 2rem 2rem;
}

<div
  className="pf-c-box__body"
>
  Box Body
</div>
`;
